src/JustSaying/Fluent/ServiceResolver/DefaultServiceResolver.cs

```diff
@@ -44,7 +45,22 @@ private object TryResolveService(Type desiredType)
         }
         else if (desiredType == typeof(IMessageSerializationFactory))
         {
+
+#if NET8_0_OR_GREATER
+            if (RuntimeFeature.IsDynamicCodeSupported)
+            {
+                #pragma warning disable IL2026
+                #pragma warning disable IL3050
+                return new NewtonsoftSerializationFactory();
+                #pragma warning restore
+            }
```

Comment:
Slightly surprised it isn't smart enough to not warn about this.

Reply:
I'll check, I think I had the check backwards when I added the `#pragma warning disable`'s

Reply:
Yeah, even with that fixed, it still warns.

Reply:
I've found an issue related to this: https://github.com/dotnet/runtime/issues/97273, looks like the supression in runtime checks just didn't make it into .NET 8, not sure if they plan to backport either.

It also looks like there are plans for a wider design proposal to support feature checks suppressing analyzers: https://github.com/dotnet/runtime/issues/96859